engine/testutil/nodes.go

```diff
@@ -112,30 +112,71 @@ func GenericNode(t testing.TB, hub *stub.Hub, identity *flow.Identity, participa
 	me, err := local.New(identity, sk)
 	require.NoError(t, err)
 
-	stubnet := stub.NewNetwork(state, me, hub)
+	stubnet := stub.NewNetwork(stateFixture.State, me, hub)
 
 	return testmock.GenericNode{
 		Log:            log,
 		Metrics:        metrics,
 		Tracer:         tracer,
+		DB:             stateFixture.DB,
+		Headers:        stateFixture.Headers,
+		Guarantees:     stateFixture.Guarantees,
+		Seals:          stateFixture.Seals,
+		Payloads:       stateFixture.Payloads,
+		Blocks:         stateFixture.Blocks,
+		Index:          stateFixture.Index,
+		State:          stateFixture.State,
+		Me:             me,
+		Net:            stubnet,
+		DBDir:          stateFixture.DBDir,
+		ChainID:        chainID,
+		ProtocolEvents: stateFixture.ProtocolEvents,
+	}
+}
+
+// CompleteStateFixture is a test helper that creates, bootstraps, and returns a StateFixture for sake of unit testing.
+func CompleteStateFixture(t testing.TB, log zerolog.Logger, metric *metrics.NoopCollector, tracer module.Tracer,
```

Comment:
We use this as a test helper to bootstrap and mutate state, as well as switch epochs. Implementing in this way enables us to inject state into the node instead of bootstrapping the state each time we were creating nodes. This makes _creating a test node that is unstaked for current epoch_ (but staked for next epoch) possible. A functionality that did not exist before.